src/main/java/uk/gov/hmcts/reform/pip/data/management/service/ValidationService.java

```diff
@@ -0,0 +1,112 @@
+package uk.gov.hmcts.reform.pip.data.management.service;
+
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Service;
+import uk.gov.hmcts.reform.pip.data.management.config.PublicationConfiguration;
+import uk.gov.hmcts.reform.pip.data.management.errorhandling.exceptions.DateValidationException;
+import uk.gov.hmcts.reform.pip.data.management.errorhandling.exceptions.EmptyRequiredHeaderException;
+import uk.gov.hmcts.reform.pip.data.management.models.publication.ArtefactType;
+
+import java.time.LocalDateTime;
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+@Service
+public class ValidationService {
+
+    private final List<String> requiredHeaders;
+    private Map<String, Object> headers;
+
+    @Autowired
+    public ValidationService() {
+        headers = new HashMap<>();
+        requiredHeaders = new ArrayList<>();
+        requiredHeaders.add(PublicationConfiguration.PROVENANCE_HEADER);
+        requiredHeaders.add(PublicationConfiguration.SOURCE_ARTEFACT_ID_HEADER);
+        requiredHeaders.add(PublicationConfiguration.TYPE_HEADER);
+    }
+
+    /**
+     * Class that guides the validation process.
+     *
+     * @param headers - a hashmap of all the headers taken in by the endpoint. Importantly, this may contain nulls (i.e.
+     *                cannot be replaced with a ConcurrentHashMap.
+     * @return Map(String, Object) - an amended version of the headers. If changes (i.e. conditional defaults)
+     *      are created, ensure the logic affects the headers map within this class.
+     */
+    public Map<String, Object> validateHeaders(Map<String, Object> headers) {
+        this.headers = headers;
```

Comment:
We should avoid setting class level variables based on the inbound request, as it has the potential to introduce issues when you have concurrent requests, both trying to update the headers at the same time.

The headers should probably be passed through in the method scope instead

Reply:
should hopefully be resolved now.